worktree.c

```diff
@@ -51,8 +51,8 @@ static struct worktree *get_main_worktree(void)
 	struct strbuf worktree_path = STRBUF_INIT;
 
 	strbuf_add_absolute_path(&worktree_path, get_git_common_dir());
-	if (!strbuf_strip_suffix(&worktree_path, "/.git"))
-		strbuf_strip_suffix(&worktree_path, "/.");
+	strbuf_strip_suffix(&worktree_path, "/.");
+ 	strbuf_strip_suffix(&worktree_path, "/.git");
```

Comment:
Sorry to keep harping on this, but the commit message will leave readers puzzled (including future me, I am sure). How about something like this instead:

```
get_main_worktree(): allow it to be called in the Git directory

When called in the Git directory of a non-bare repository, this function
would not return the directory of the main worktree, but of the Git
directory instead.

The reason: when the Git directory is the current working directory, the
absolute path of the common directory will be reported with a trailing
`/.git/.`, which the code of `get_main_worktree()` does not handle
correctly.

Let's fix this.
```

While writing this commit message, I realized that the proposed patch introduces a regression: if `worktree_path` ends in `/./.git` at this stage, the original code would treat it correctly while the patched code wouldn't.

How about keeping the original code, and just insert `strbuf_strip_suffix(&worktree_path, "/.");` before the `if`?

Reply:
I do not see how this was resolved :-(